thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -755,8 +764,13 @@ class ThinReplicaImpl {
           auto num_events = kvb_update.kv_pairs.size();
           auto client_id = getClientId(context);
           LOG_DEBUG(logger_, "Sending updates (history, data)" << KVLOG(client_id, block_id, num_events));
-
           sendData(stream, kvb_update);
+          metrics.last_sent_block_id.Get().Set(block_id);
```

Comment:
Instead of updating this metric here and in [SubscribeToUpdates #L441](https://github.com/vmware/concord-bft/blob/72cfb4761fd6a7236b5a9a21b354c1c4f4775ce4/thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp#L441), why don't we update the metric in `SendData`/`SendHash` methods? If for some reason stream is closed when writing data over the stream, then we would still persist the correct value of last_sent_block_id, and not be off by 1. WDYT?

Reply:
We throw exception when we stream is found to be closed. In such case, remaining code incrementing metric won't be executed.

Reply:
Yes, thanks, I missed that.